src/lava/lib/peripherals/liveplot/liveplot.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import cv2
+import numpy as np
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.ports.ports import InPort
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+
+import time
+
+
+class OpenCVDisplay(AbstractProcess):
+    """Process that receives datas from connected processmodel.
+    The frame data which are needed to display from connected processmodels
+    need a corresponding InPort which is connected to an OutPort.
+    The number of input ports is decided by the number of
+    process models connected to this opencv process model.
+    """
+    def __init__(self,
+                 shape):
```

Comment:
It is not apparent what would be the expected dimentionality of the `shape` here.
Should it be 2D ? (like (125, 125) ?)
Should it be 3D, with last dimension being 1 (for grayscale images) ? (like (125, 125, 1) ?)
Should it be 3D, with last dimension being 3 (for RGB images) ? (like (125, 125, 3) ?)

In any case, `shape` parameter should be validated in the `__init__` method so that invalid formats of `shape` throw an error from the start.

Reply:
2D and 3D for grayscale, 3D for RGB. Added the explanation in the docstring